### PR TITLE
New modal design

### PR DESCRIPTION
**Pull request by Tanza3D**
bye old one

### CODE REVIEW

**Comment by Coppertine** [comment]
Should add in a functionality that detects clicks outside of the modal which would in turn call the cancel callback.

**Comment by Tanza3D** [comment]
will do :+1:

File: global/js/main.js [hunk 1]
```diff
@@ -707,19 +724,19 @@ document.addEventListener("DOMContentLoaded", function () {
...
 
     var oTabContainers = document.querySelectorAll("[otab-container]");
-    for(let oTabContainer of oTabContainers) {
+    for (let oTabContainer of oTabContainers) {
         let oTabs = oTabContainer.querySelectorAll("[otab-name]")
         for (let oTab of oTabs) {
             if (oTab.getAttribute("otab-default") == "") {
                 oTab.classList.remove("osekai__otab-hidden");
-                oTabContainer.querySelector("[otab-button=\""+oTab.getAttribute("otab-name")+"\"]").classList.add("osekai__otab-button-active")
+                oTabContainer.querySelector("[otab-button=\"" + oTab.getAttribute("otab-name") + "\"]").classList.add("osekai__otab-button-active")
```
Comment (Coppertine) [inline global/js/main.js]:
```suggestion
                oTabContainer.querySelector(`[otab-button="${oTab.getAttribute("otab-name")}"]`).classList.add("osekai__otab-button-active")

```

File: snapshots/js/functions.js [hunk 2]
```diff
@@ -117,12 +118,12 @@ async function loadVersion(vername, pushstate = true, fromButton = false) {
 
     var reldate = new Date(thisver["version_info"]["release"] * 1000).toLocaleDateString("en-AU", { month: "long", day: "numeric", year: "numeric" });
     document.getElementById("release_date").innerHTML = await GetStringRaw("snapshots", "version.released", [reldate]);
-    
 
 
-    document.getElementById("archived_by").innerHTML = await GetStringRaw("snapshots", "version.archivedBy", [`<strong class="user_hover" userid="` + thisver["archive_info"]["archiver"] + `" hoverside="right">` + thisver["archive_info"]["archiver"] + `</strong>`]); 
 
-    if(thisver["archive_info"]["archiver_id"] != null && thisver["archive_info"]["archiver_id"] != ""){
+    document.getElementById("archived_by").innerHTML = await GetStringRaw("snapshots", "version.archivedBy", [`<strong class="user_hover" userid="` + thisver["archive_info"]["archiver"] + `" hoverside="right">` + thisver["archive_info"]["archiver"] + `</strong>`]);
```
Comment (Coppertine) [inline snapshots/js/functions.js]:
```suggestion
    document.getElementById("archived_by").innerHTML = await GetStringRaw("snapshots", "version.archivedBy", [`<strong class="user_hover" userid="${thisver["archive_info"]["archiver"]}`" hoverside="right">${thisver["archive_info"]["archiver"]}</strong>`]);

```

**Review by Coppertine — CHANGES_REQUESTED** [review]
Alright so far, although would need to fix z-indexing and include other methods to close a modal.

**Comment by Tanza3D** [comment]
my issue is that sometimes a modal *shouldn't* be closeable, but i cant think of any situations that'd be a thing as of right now atleast, so for now i'll just add more ways of closing them out and we can implement a way to disable them in the future

**Comment by Tanza3D** [comment]
fixed light mode

**Review by Coppertine — CHANGES_REQUESTED** [review]
Text is not coloured correctly on light mode and looks unreadable when using Dialog Test button. Lightmode looks good on the report dialogs. But not the Dialog Test.

**Comment by Tanza3D** [comment]
i love light mode... will fix

**Comment by Tanza3D** [comment]
i think fixed...? @Coppertine

**Comment by Coppertine** [comment]
image
Buttons not looking well on lightweight
image

**Comment by Tanza3D** [comment]
should be a separate issue i think, seems like a different thing

File: global/js/comment_system.js [hunk 3]
```diff
@@ -359,9 +370,20 @@ function openReply(strCommentId, element) {
 
         document.getElementById("reply__send").addEventListener("click", () => {
             if (document.getElementById("reply__input").value.includes("https://osu.ppy.sh/beatmapsets/")) {
-                openDialog("Are you sure you want to post this comment?", "Your reply looks like it contains an osu! beatmap URL.", "If you're trying to post a beatmap, you should instead use the 'Add' button on the beatmaps panel.", "Post Anyway", function () {
-                    replySend();
-                });
+                openDialog("Are you sure you want to post this comment?", "Your reply looks like it contains an osu! beatmap URL.", "If you're trying to post a beatmap, you should instead use the 'Add' button on the beatmaps panel.", [
```
Comment (Coppertine) [inline global/js/comment_system.js:373]: Would this be replaced with a translation key?

**Review by Coppertine — APPROVED** [review]
LGTM

**Comment by Tanza3D** [comment]
https://github.com/Osekai/osekai-wip/pull/50

File: global/js/comment_system.js (hunk 3 above)
Comment (Tanza3D) [reply global/js/comment_system.js:373]: nah, would just be passed as string from a function, best to do that if possible in most places

**Pull request merged**